src/types.ts

```diff
@@ -615,7 +660,11 @@ export interface ChatCompletion {
     message: {
       name?: string;
       role: "system" | "assistant" | "user";
-      content: string;
+      content: string | null;
```

Comment:
this is a somewhat "breaking" change, but with the introductions of functions, seems like this value can now be `null`

Reply:
This feels really bad to me. hate to have to tell typescript to ignore the null every single time I need to extract the content.

I feel like we could do more intelligent typing by looking at the `functions` and `function_call` parameter to see if there is a possibility of the response being null. Would you be willing to try doing this?

Reply:
ya, I had the same thought.  Shouldn't be too hard to compose the types to get it to work that way

Reply:
I think part of the philosophy of this library is to try to pass the raw response as often as possible. Transforming the response even for something as minor as this would mean that people couldn't rely on open ai docs. The complex typings look nice, I'd accept this PR if it were to be added.

Reply:
I actually don't think this is a great solution...

1. The conditional type is a PITA because the property is nested so deep.  It will likely make the return a lot harder to understand.  There may be an easier way I'm unaware of..
```ts
  async createChatCompletion<T extends ChatCompletionOptions>(
    options: T
  ): Promise<
    T extends { functions: any }
      ? Omit<ChatCompletion, "choices"> & {
          choices: Omit<ChatCompletionChoice, "message"> & {
            message: Omit<ChatCompletionChoiceMessage, "content"> & {
              content: string | null;
            };
          };
        }
      : ChatCompletion
  > {
```

2. It's not bulletproof.  There are many instances where typescript may not know if the input has the function's property.  In those cases, you've lost the safety of "strict null checks` on content.  This will almost certainly cause NPE for someone.

IMO, the "null coalesce" to `""` is the superior solution, but I respect whatever you want to do.  

I may actually abandon the PR anyway, because I'm having issues with the error handling (`throwError`) of this library swallowing status code and other raw request details.

Reply:
> IMO, the "null coalesce" to "" is the superior solution, but I respect whatever you want to do.

Thanks for the insight. I see where you're coming from on the typing front. Maybe worth revisiting in the future, but after further pondering I think a "null coalesce" is more than reasonable.

> I may actually abandon the PR anyway, because I'm having issues with the error handling (`throwError`) of this library swallowing status code and other raw request details.

Error handling was kind of hacked in after the fact 😅. If you have any ideas on how to improve it, I'm all ears!

Reply:
I'd probably copy axios and do something like this:
```ts
class HttpResponseError extends Error {
  httpResponse: Response;

  constructor(message: string, httpResponse: Response) {
    super(message);

    // Set the prototype explicitly to allow instanceof checks
    Object.setPrototypeOf(this, HttpResponseError.prototype);

    this.name = 'HttpResponseError';
    this.httpResponse = httpResponse;
  }
}
```

Or maybe even just have properties for`status`, `body`, etc if it's already been parsed.

Then you can do something like this in your catch blocks:

```ts
}catch(e){
  if(e instanceof HttpResponseError){
     if(e.statusCode === 429){
        // retry request with backoff
        return retry();
     }
  }
  throw e;
}
```

Reply:
Sounds good, thanks for the issue.